python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll

```diff
@@ -319,6 +328,42 @@ class KwUnpacked extends Node, TKwUnpacked {
   override Location getLocation() { result = call.getLocation() }
 }
 
+/**
+ * A synthetic node representing an iterable sequence. Used for changing content type
+ * for instance from a `ListElement` to a `TupleElement`.
+ */
+class IterableSequence extends Node, TIterableSequence {
+  SequenceNode consumer;
+
+  IterableSequence() { this = TIterableSequence(consumer) }
+
+  override string toString() { result = "IterableSequence" }
+
+  override DataFlowCallable getEnclosingCallable() {
+    result = any(CfgNode node | node = TCfgNode(consumer)).getEnclosingCallable()
+  }
+
+  override Location getLocation() { result = consumer.getLocation() }
+}
+
+/**
+ * A synthetic node representing an iterable element. Used for changing content type
+ * for instance from a `ListElement` to a `TupleElement`.
```

Comment:
This example is exactly the same as for `IterableSequence`, which may make it unclear why both classes are needed.

Reply:
Right, I could add something like "via a read step to an IterableElement followed by a store step" to the above and "usually via a read step from an IterableSequence followed by a store step" here.